### PR TITLE
feat(switch): ✨  add switch component

### PR DESCRIPTION
**Pull request by navin-moorthy**
Switch Component follows the same API as Checkbox.

### CODE REVIEW

File: src/switch/Switch.tsx [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+
+import { forwardRefWithAs } from "../utils/types";
+import { Box, BoxProps } from "../box";
+import { useTheme } from "../theme";
+
+export type SwitchStatus = CheckboxStateReturn["state"];
+
+export type SwitchStateValues = {
+  state?: SwitchStatus;
+  size?: "xs" | "sm" | "lg";
```
Comment (anuraghazra) [inline src/switch/Switch.tsx:17]: Use Renderlesskit.GetThemeValue

File: src/theme/defaultTheme/index.ts [hunk 2]
```diff
@@ -22,6 +24,7 @@ const theme = {
   iconButton,
   spinner,
   tag,
+  switch: Switch,
```
Comment (anuraghazra) [inline src/theme/defaultTheme/index.ts:27]: use lowercase switch

File: src/switch/Switch.tsx [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+export type SwitchStateProps = SwitchStateValues & SwitchStateActions;
+
+export type SwitchLabelProps = BoxProps & SwitchStateProps;
+
+export const SwitchLabel = forwardRefWithAs<
+  SwitchLabelProps,
+  HTMLLabelElement,
+  "label"
+>((props, ref) => {
+  const { state, setState, size, ...mainProps } = props;
```
Comment (anuraghazra) [inline src/switch/Switch.tsx:31]: Any use of state, setState here 🤔

File: src/theme/defaultTheme/index.ts (hunk 2 above)
Comment (navin-moorthy) [reply src/theme/defaultTheme/index.ts:27]: lowercase switch is reserved keyword, may be `_switch` is okay.

File: src/switch/Switch.tsx (hunk 3 above)
Comment (navin-moorthy) [reply src/switch/Switch.tsx:31]: I will change these back to context, seems not nice.

File: src/theme/defaultTheme/index.ts (hunk 2 above)
Comment (anuraghazra) [reply src/theme/defaultTheme/index.ts:27]: Ohhhhhhhhhhh yeah yeah.

File: src/switch/Switch.tsx [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+});
+
+export type SwitchProps = BoxProps & {
+  state?: SwitchStateContext["state"];
+  defaultState?: SwitchStateContext["state"];
+  onStateChange?: (value: SwitchStateContext["state"]) => void;
+  size?: SwitchThemeContext["size"];
+};
+
+export const Switch: React.FC<SwitchProps> = props => {
```
Comment (anuraghazra) [inline src/switch/Switch.tsx:132]: Typesafe render props

```ts
type SwitchRenderProps = {
  children?:
    | (({
        state,
        theme,
      }: {
        state: SwitchStateContext;
        theme: SwitchThemeContext;
      }) => JSX.Element)
    | React.ReactNode;
};

export const Switch: React.FC<SwitchProps & SwitchRenderProps> = props => {
...
```

File: src/switch/stories/Switch.stories.tsx [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+export const CustomSwitch = storyTemplate<SwitchProps>(
+  args => (
+    <Switch {...args}>
+      {({
+        state: { state },
+        theme,
+      }: {
+        state: SwitchStateContext;
+        theme: SwitchThemeContext;
+      }) => {
```
Comment (anuraghazra) [inline src/switch/stories/Switch.stories.tsx:59]: We can remove this typings after making the typesafe render prop.

**Review by anuraghazra — APPROVED** [review]

**Pull request merged**